cdap-watchdog/src/main/java/co/cask/cdap/logging/plugins/LocationManager.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.logging.plugins;
+
+import co.cask.cdap.common.io.Locations;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.io.Closeables;
+import org.apache.twill.filesystem.Location;
+import org.apache.twill.filesystem.LocationFactory;
+
+import java.io.BufferedReader;
+import java.io.Closeable;
+import java.io.Flushable;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Manage locations for {@link RollingLocationLogAppender}
+ */
+public class LocationManager implements Flushable, Closeable {
+  protected static final String TAG_NAMESPACE_ID = ".namespaceId";
+  protected static final String TAG_APPLICATION_ID = ".applicationId";
+
+  private Location logBaseDir;
+  private Map<LocationIdentifier, LocationOutputStream> activeLocations;
+  private String filePermissions;
+
+  public LocationManager(LocationFactory locationFactory, String basePath, String filePermissions)
+    throws IOException {
+    this.logBaseDir = locationFactory.create(basePath);
+    this.activeLocations = new HashMap<>();
+    this.filePermissions = filePermissions;
+  }
+
+  LocationIdentifier getLocationIdentifier(Map<String, String> propertyMap) throws IllegalArgumentException,
```

Comment:
`IllegalArgumentException` is a runtime exception, no need to declare it in `throws`. Also, it's good to have javadoc for non-private method to include what this method does, what's the param and what exceptions it will throw under what condition.